packages/vm/src/modules/file_system_cache.rs

```diff
@@ -65,6 +65,42 @@ use super::CachedModule;
 ///   New version because of Metering middleware change.
 const MODULE_SERIALIZATION_VERSION: &str = "v10";
 
+/// Function that actually does the heavy lifting of creating the module version discriminator.
+///
+/// Separated for sanity tests because otherwise the `OnceLock` would cache the result.
+#[inline]
+fn raw_module_version_discriminator() -> String {
+    let hashes = [COST_FUNCTION_HASH];
+
+    let mut hasher = blake3::Hasher::new();
+
+    hasher.update(MODULE_SERIALIZATION_VERSION.as_bytes());
+    hasher.update(wasmer::VERSION.as_bytes());
+
+    for hash in hashes {
+        hasher.update(hash);
+    }
+
+    hasher.finalize().to_hex().to_string()
+}
+
+/// Bump this version whenever the module system changes in a way
+/// that old stored modules would be corrupt when loaded in the new system.
+/// This needs to be done e.g. when switching between the jit/native engine.
+///
+/// The string is used as a folder and should be named in a way that is
+/// easy to interpret for system admins. It should allow easy clearing
+/// of old versions.
```

Comment:
Is this still the case though?

Reply:
What I meant with my comment here is that the hash is no longer "easy to interpret for system admins" and also doesn't "allow easy clearing of old versions". That's the tradeoff we're making here, which is fine imho. Sysadmins can just check what folder the latest file was created in.
But that means, we should remove this from the docs